services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,43 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  xQueueSend(thermalMgrQueueHandle, event, (TickType_t) 10);
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t osEvent = {0};
+  osEvent.type = THERMAL_MGR_EVENT_OS_CMD;
+
+  thermalMgrSendEvent(&osEvent);
 }
 
-static void thermalMgr(void *pvParameters) {
+static void thermalMgr(void *pvParameters) { // what is pvParameters used for?
   /* Implement this task */
+  
+  thermal_mgr_event_t queueEventItemBuffer = {0};
+
   while (1) {
     
+    if (xQueueReceive(thermalMgrQueueHandle, &queueEventItemBuffer, (TickType_t) 10) == pdPASS) {
+      
+      float tempC;
+      readTempLM75BD(LM75BD_OBC_I2C_ADDR, &tempC);
```

Comment:
- [ ] only read when an event tells you too. Let's say we get an event that doesn't require reading then we would read when we shouldn't have
- [ ] in general, check and log the return error code from non void functions